torch/quantization/fx/quantize.py

```diff
@@ -0,0 +1,727 @@
+    def convert(self, quantizer, node):
+        assert self.all_tensors
+        return quantize(quantizer, node)
+
+# 2. Post Training Dynamic Quantizatoin Patterns
+@register_dynamic_pattern(torch.nn.Linear)
+@register_dynamic_pattern(torch.nn.functional.linear)
+class DynamicLinear(QuantizerPattern):
+     def __init__(self, quantizer, node):
+        super().__init__(quantizer, node)
+        self.linear_node = node
+        if node.op == 'call_module':
+            assert isinstance(quantizer.modules[node.target], torch.nn.Linear)
+            self.linear = quantizer.modules[self.linear_node.target]
+
+     def convert(self, quantizer, node, load_arg, debug=False):
+        if self.linear_node.op == 'call_module':
+            quantized = torch.nn.quantized.dynamic.Linear.from_float(self.linear)
+            parent_name, name = _parent_name(self.linear_node.target)
+            setattr(quantizer.modules[parent_name], name, quantized)
+            return quantizer.quantized_graph.call_module(self.linear_node.target, (load_arg(quantized=False)(self.linear_node.args[0]),), {})
+        elif self.linear_node.op == 'call_function':
+            if debug:
+                # quantize and dequantize weight
+                args = load_arg(quantized=[1])(self.linear_node.args)
+                args = load_arg(quantized=False)(self.linear_node.args)
+                kwargs = load_arg(quantized=False)(self.linear_node.kwargs)
+                return quantizer.quantized_graph.call_function(torch.nn.functional.linear, args, kwargs)
+            else:
+                # quantize and dequantize weight
+                args = load_arg(quantized=[1])(self.linear_node.args)
+                kwargs = load_arg(quantized=False)(self.linear_node.kwargs)
+                # pack weight
+                weight = load_arg(quantized=True)(self.linear_node.args[1])
+                bias = None
+                other_args = load_arg(quantized=False)(self.linear_node.args[1:])
+                if len(self.linear_node.args) > 2:
+                    bias = load_arg(quantized=False)(self.linear_node.args[2])
+                    other_args = other_args[1:] # remove the bias argument
+                else:
+                    assert 'bias' in kwargs, \
+                        'expect bias provided as a keyword argument when it is not a positional argument'
+                    bias = kwargs['bias']
+                    kwargs.pop('bias')
+                prepack_args = [weight, bias]
+                packed_weight = quantizer.quantized_graph.call_function(torch.ops.quantized.linear_prepack, prepack_args, {})
+                # construct dynamic linear input
+                linear_input = load_arg(quantized=False)(self.linear_node.args[0])
+                qdynamic_linear_args = [linear_input, packed_weight]
+                return quantizer.quantized_graph.call_function(torch.ops.quantized.linear_dynamic, qdynamic_linear_args, kwargs)
+
+class Quantizer:
```

Comment:
since these pattern also contain implementations of the feature I feel it might make more sense to put them together. see reference to `quantizer` in the `convert` function of the pattern. the pattern can be considered to be 'tied' to the quantizer implementation.

Reply:
Should we consider calling it something other than `Quantizer`? Sine the quantization backend currently uses it for creating different types of qtensors.

Reply:
yeah, I think so. the official API is not figured out yet, I think maybe we'll not expose this to user and have a quantize_fx as a top level function in the end.

Reply:
Second comments here: Request moving the quantizer to a separate file and calling it something else other than Quantizer.